kinematics_base_test/src/test_kinematics_plugin.cpp

```diff
@@ -100,29 +91,33 @@ class KinematicsTest
   }
 
   void searchIKCallback(const geometry_msgs::Pose &ik_pose,
-                            const std::vector<double> &joint_state,
-                            moveit_msgs::MoveItErrorCodes &error_code)
-  {
+                        const std::vector<double> &joint_state,
+                        moveit_msgs::MoveItErrorCodes &error_code) {
     std::vector<std::string> link_names;
     link_names.push_back(tip_link_);
     std::vector<geometry_msgs::Pose> solutions;
     solutions.resize(1);
-    if(!kinematics_solver_->getPositionFK(link_names,joint_state,solutions))
-    {
+    if (!kinematics_solver_->getPositionFK(link_names, joint_state,
+                                           solutions)) {
       error_code.val = error_code.PLANNING_FAILED;
       return;
     }
 
-    EXPECT_GT(solutions[0].position.z,0.0f);
-    if(solutions[0].position.z > 0.0)
+    EXPECT_GT(solutions[0].position.z, 0.0f);
+    if (solutions[0].position.z > 0.0)
       error_code.val = error_code.SUCCESS;
     else
       error_code.val = error_code.PLANNING_FAILED;
   };
 
-
 public:
+  // Definitions for handling smart pointers in Kinetic
+  template <typename T> constexpr T &lvalue(T &&r) noexcept { return r; }
```

Comment:
This converts the rvalue to lvalue before it is fed to make_shared_ptr. A good reference for this that I found for this is [lvalues and rvalues](http://eli.thegreenplace.net/2011/12/15/understanding-lvalues-and-rvalues-in-c-and-c)